### PR TITLE
Juno extra kernel args

### PR DESCRIPTION
**Pull request by roxell**

### CODE REVIEW

**Comment by mwasilew** [comment]
You could do sth like this on top of your patch:
```
$ git diff
diff --git a/projects/lkft/devices/juno-r2 b/projects/lkft/devices/juno-r2
index c260eef..f74e4a9 100644
--- a/projects/lkft/devices/juno-r2
+++ b/projects/lkft/devices/juno-r2
@@ -1,8 +1,9 @@
 {% extends "devices/juno-r2" %}
 
+{% set extra_kernel_args = "default_hugepagesz=2M hugepages=256" %}
+
 {% block context %}
   {{ super() }}
   booti_dtb_addr: "0x85000000"
-  extra_kernel_args: 'default_hugepagesz=2M hugepages=256'
   extra_nfsroot_args: ',wsize=65536'
 {% endblock context %}
diff --git a/testcases/master/template-kselftest.yaml.jinja2 b/testcases/master/template-kselftest.yaml.jinja2
index ac9f3fc..2131747 100644
--- a/testcases/master/template-kselftest.yaml.jinja2
+++ b/testcases/master/template-kselftest.yaml.jinja2
@@ -14,7 +14,7 @@
   {#- If vsyscall_mode is defined, pass it as an extra kernel argument #}
   {#- vsyscall_mode may be emulate (default), native, or none.         #}
   {%- if vsyscall_mode is defined %}
-  extra_kernel_args: 'vsyscall={{vsyscall_mode}}'
+  extra_kernel_args: 'vsyscall={{vsyscall_mode}} {{extra_kernel_args|default("")}}'
   {%- endif %}
 {% endblock context %}
 
diff --git a/testcases/master/template-kvm-unit-tests.yaml.jinja2 b/testcases/master/template-kvm-unit-tests.yaml.jinja2
index ae77b3c..ce04c9c 100644
--- a/testcases/master/template-kvm-unit-tests.yaml.jinja2
+++ b/testcases/master/template-kvm-unit-tests.yaml.jinja2
@@ -4,7 +4,7 @@
 
 {% block context %}
 {{ super() }}
-  extra_kernel_args: 'kvm.enable_vmware_backdoor=1 kvm.force_emulation_prefix=1'
+  extra_kernel_args: 'kvm.enable_vmware_backdoor=1 kvm.force_emulation_prefix=1 {{extra_kernel_args|default("")}}'
 {% endblock context %}
 
 {% set test_timeout = 25 %}
```

**Comment by roxell** [comment]
great suggestion.

**Comment by mwasilew** [comment]
It won't work on it's own as you require the extra args for all juno tests but at least gives an example.

File: testcases/master/template-kvm-unit-tests.yaml.jinja2 [hunk 1]
```diff
@@ -2,10 +2,7 @@
 
 {% set use_context = true %}
 
-{% block context %}
-{{ super() }}
-  extra_kernel_args: 'kvm.enable_vmware_backdoor=1 kvm.force_emulation_prefix=1'
-{% endblock context %}
+{% set extra_kernel_args = 'kvm.enable_vmware_backdoor=1 kvm.force_emulation_prefix=1' + extra_kernel_args|default("") %}
```
[the remark this answers is not in the document]
Comment (mwasilew) [reply testcases/master/template-kvm-unit-tests.yaml.jinja2]: My bad, this is outside of the block.

**Pull request merged**